### PR TITLE
PullRequest Cosmetic Fixes [2020-01-07]

### PR DESCRIPTION
**Pull request by PullRequest-Agent**
The following proposed changes fix a collection of trivial, surface-level issues raised by PullRequest reviewers while conducting a thorough review of the DAGMC project (Full review to follow).

The intent of this pull request is to address the non-functional fixes discovered so far and save the maintainers of the DAGMC project the time it would require to address them individually in the future.

An approximate breakdown of issue types this pull request will resolve follows:

Issue Type | Approx. Count
------------ | -------------
Typo / Grammar | 55
Unused Import | 1
Unused variable | 3
External link corrections | 21
Misc readability | 7
**TOTAL** | **87**

_NOTE: The collection of fixes in this pull request are not necessarily exhaustive. It's possible and likely a number of additional like-issues, largely trivial in nature, will be surfaced as the PullRequest team nears completion of the project's full and comprehensive review and additional pull requests will be created._ 

<p align="right">
    <a href="https://www.pullrequest.com/">
        
    </a>
</p>

### CODE REVIEW

File: src/overlap_check/overlap.cpp [hunk 1]
```diff
@@ -108,7 +108,6 @@ check_instance_for_overlaps(std::shared_ptr<Interface> MBI,
 
       EntityHandle vert = all_verts[i];
       CartVect loc;
-      rval = MBI->get_coords(&vert, 1, loc.array());
```
Comment (gonuke) [inline src/overlap_check/overlap.cpp]: Although the value of `rval` is not checked, this call is necessary to populate `loc` correctly.

File: src/uwuw/uwuw.cpp [hunk 2]
```diff
@@ -151,7 +151,6 @@ int UWUW::get_length_of_table(std::string filename, std::string datapath) {
   hid_t arr_space = H5Dget_space(ds);
 
   hsize_t arr_dims[1];
-  int arr_ndim = H5Sget_simple_extent_dims(arr_space, arr_dims, NULL);
```
Comment (gonuke) [inline src/uwuw/uwuw.cpp]: although the return value `arr_ndim` is not used, this call is necessary to populate `arr_dims`

**Review by gonuke — CHANGES_REQUESTED** [review]
Thanks for this thorough review and the many small but valuable changes.

I see two changes that I think are semantically changing the code in ways that are not correct.

**Comment by ljacobson64** [comment]
I briefly looked over this PR and did not notice anything other than the issues Paul brought up in his comments that I thought would cause problems.

**Pull request merged**